action/move.php

```diff
@@ -8,56 +8,186 @@
+use dokuwiki\plugin\struct\types\Page;
 
 if(!defined('DOKU_INC')) die();
 
 class action_plugin_struct_move extends DokuWiki_Action_Plugin {
 
+    /** @var helper_plugin_sqlite */
+    protected $db = null;
+
     /**
      * Registers a callback function for a given event
      *
      * @param Doku_Event_Handler $controller DokuWiki's event controller object
      * @return void
      */
     public function register(Doku_Event_Handler $controller) {
-        $controller->register_hook('PLUGIN_MOVE_PAGE_RENAME', 'AFTER', $this, 'handle_move');
+        $controller->register_hook('PLUGIN_MOVE_PAGE_RENAME', 'AFTER', $this, 'handle_move', true);
+        $controller->register_hook('PLUGIN_MOVE_MEDIA_RENAME', 'AFTER', $this, 'handle_move', false);
     }
 
     /**
      * Renames all occurances of a page ID in the database
      *
      * @param Doku_Event $event event object by reference
-     * @param mixed $param [the parameters passed as fifth argument to register_hook() when this
-     *                           handler was registered]
+     * @param bool $ispage is this a page move operation?
      * @return bool
      */
-    public function handle_move(Doku_Event $event, $param) {
+    public function handle_move(Doku_Event $event, $ispage) {
         /** @var helper_plugin_struct_db $hlp */
         $hlp = plugin_load('helper', 'struct_db');
-        $db = $hlp->getDB(false);
-        if(!$db) return false;
+        $this->db = $hlp->getDB(false);
+        if(!$this->db) return false;
         $old = $event->data['src_id'];
         $new = $event->data['dst_id'];
 
-        // ALL data tables (we don't trust the assigments are still there)
+        // prepare work
+        $this->db->query('BEGIN TRANSACTION');
+
+        // general update of our meta tables
+        if($ispage) {
+            $this->updateDataTablePIDs($old, $new);
+            $this->updateAssignments($old, $new);
+            $this->updateTitles($old, $new);
+        }
+
+        // apply updates to all columns in all schemas depending on type
+        $schemas = Schema::getAll();
+        foreach($schemas as $table) {
+            $schema = new Schema($table);
+            foreach($schema->getColumns() as $col) {
+                if($ispage) {
+                    switch(get_class($col->getType())) {
+                        case Page::class:
+                            $this->updateColumnID($schema, $col, $old, $new, true);
+                            break;
+                        case Lookup::class:
+                            $this->updateColumnLookup($schema, $col, $old, $new);
+                            break;
+                    }
+                } else {
+                    switch(get_class($col->getType())) {
+                        case Media::class:
+                            $this->updateColumnID($schema, $col, $old, $new);
+                            break;
+                    }
+                }
+            }
+        }
```

Comment:
I'm not too happy with this loop. It has a lot of nesting levels, but I'm not sure how to make better. Suggestions?